jetty-server/src/test/java/org/eclipse/jetty/server/HttpOutputTest.java

```diff
@@ -767,16 +767,23 @@ public void setNext(Interceptor interceptor)
     @Test
     public void testEmptyArray() throws Exception
     {
-        AtomicBoolean committed = new AtomicBoolean();
+        FuturePromise<Boolean> committed = new FuturePromise<>();
```

Comment:
If you're using `FuturePromise` now, maybe you should add a timeout for the `get()` call.

Reply:
oops I missed this comment.  I'm planning for success and the test will pass!        If we break something, then a test that hangs forever will get just as much attention than one that times out.

Reply:
Yeah but its not ideal, sometimes its hard to find where it timed out and why. Much better to just fail after a few seconds.

Reply:
I've fixed this in the 10 PR which has not yet been merged.